sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java

```diff
@@ -60,6 +66,11 @@ class NetworkInterfaceImpl
     private NetworkSecurityGroup existingNetworkSecurityGroupToAssociate;
     /** cached related resources. */
     private NetworkSecurityGroup networkSecurityGroup;
+    /** the delete option of public ip address */
+    private DeleteOptions publicIpAddressDeleteOption;
+    /** the name of specified ip config name */
+    private Map<DeleteOptions, Set<String>> specifiedIpConfigNames;
+    private boolean isInUpdateIpConfigMode = false;
```

Comment:
These properties seems not cleared after creating/updating. It'll cause issue for next update.

Reply:
1. The property `publicIpAddressDeleteOption` has been deprecated and deleted in the new version.
2. The above properties are initialized in the additional [overloaded method `update`](https://github.com/Azure/azure-sdk-for-java/pull/39096/files#diff-37c578353b996b193453fda4edc3ea63b50e780c0e21ae5a33d6c4a8a8e25ceeR618) before update.
3. Append `this.specifiedIpConfigNames.clear();` for the method `beforeCreating` in the new version.

Reply:
1. These properties cleard in the overloaded method `update` when update.
2. The `publicIpAddressDeleteOption` is deprecated.
3. Add init process for these properties in the constructor of the `NetworkInterfaceImpl` class.